app/src/main/java/com/andela/art/login/LoginActivity.java

```diff
@@ -0,0 +1,175 @@
+package com.andela.art.login;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.Toast;
+
+import com.andela.art.serialentry.SerialEntryActivity;
+import com.andela.art.R;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GoogleAuthProvider;
+import com.shobhitpuri.custombuttons.GoogleSignInButton;
+
+import uk.co.chrisjenx.calligraphy.CalligraphyConfig;
+import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;
+
+
+public class LoginActivity extends AppCompatActivity implements LoginActivityMVP.LoginActivity {
+
+    @BindView(R.id.google_sign_in_button)
+    GoogleSignInButton button;
+    FirebaseAuth mAuth;
+    private final static int RC_SIGN_IN = 2;
+    GoogleSignInClient mGoogleSignInClient;
+    private static final String TAG = "LoginActivity";
+    FirebaseAuth.AuthStateListener mAuthListener;
+    // A progress dialog to display when the user is connecting in
+    // case there is a delay in any of the dialogs being ready.
+    private ProgressDialog mConnectionProgressDialog;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mAuth.addAuthStateListener(mAuthListener);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (mAuthListener != null) {
+            mAuth.removeAuthStateListener(mAuthListener);
+        }
+    }
+
+    @Override
+    protected void attachBaseContext(Context newBase) {
+        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        CalligraphyConfig.initDefault(new CalligraphyConfig.Builder()
+                .setDefaultFontPath("fonts/Roboto-Regular.ttf")
+                .setFontAttrId(R.attr.fontPath)
+                .build()
+        );
+        setContentView(R.layout.google_login);
```

Comment:
This could point to `activity_login` xml. It would be more intituive and that xml would be easy to find. Plus I think it's convention to do it like that. https://hungsonandroid.wordpress.com/2016/10/17/androidnaming-convention/